src/HotChocolate/Core/src/Types/Types/InputParser.cs

```diff
@@ -37,6 +37,15 @@ public InputParser(ITypeConverter converter)
 
             var runtimeValue = ParseLiteralInternal(value, field.Type, field.Name, 0, true, field);
             runtimeValue = FormatValue(field, runtimeValue);
+
+            // Caller doesn't care, but to ensure specificity, we set the field's runtime type
+            // to make sure it's at least converted to the right type.
+            // e.g. from a list to an array if it should be an array
+            if (targetType == typeof(object))
```

Comment:
this expression can also include the expression below ... targetType == null || is typeof(object)